### PR TITLE
Allow grouping bar chart series into "Other" category

### LINKED ISSUE #11325: Allow aggregating values of series breakout into "Other" category
When using a series breakout where the variable has many distinct values, plots can be cluttered and difficult to read.

I would like to be able to say something like "Plot the top k groups (in terms of the variable set as the Y axis) and aggregate the remaining groups into "Other". 

Most of my queries are written directly in SQL, and It is possible to do this aggregation in the code, but it is cumbersome, repetitive, and error-prone.

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

### LINKED ISSUE #41986: Allow for grouping large category counts into "other"
PRD: https://www.notion.so/metabase/Allow-for-grouping-large-category-counts-into-other-61660fc5206a48ed8ce12a19c2e6ef96

### PR DESCRIPTION
**Pull request by kulyk**
Closes #41986, closes #11325
[Product doc](https://www.notion.so/metabase/Allow-for-grouping-large-category-counts-into-other-61660fc5206a48ed8ce12a19c2e6ef96)

Makes it possible to group a certain number of series into an "Other" series for bar charts

### To verify

1. New > Question > Orders > Count by Created At and User.State > Visualize > Make it a bar chart
2. Open the visualization settings sidebar and click on the ellipsis icon next to the "User — State" column
3. Click on the "Enforce maximum number of series" checkbox
4. Ensure some series are now grouped into "Other"
5. Change the number below the checkbox and make sure the changes are reflected on the chart
6. Ensure you can move the columns in and out of the "Other" series using the viz settings sidebar
7. Ensure the "Other" series behaves like a regular series (you can toggle visibility, it gets highlighted on hover, etc.)

### Demo

https://github.com/user-attachments/assets/de6e5be7-fabb-4cb3-aa85-3e937e3fba08

### CODE REVIEW

File: frontend/src/metabase-types/api/card.ts
Comment (alxnddr) [inline frontend/src/metabase-types/api/card.ts:154]: It would be nice to add `graph.max_categories_enabled`

File: frontend/src/metabase/visualizations/visualizations/CartesianChart/events.ts
Comment (alxnddr) [inline frontend/src/metabase/visualizations/visualizations/CartesianChart/events.ts:489]: What if we could have this series values right in `chartModel.dataset[OTHER_DATA_KEY]`? It seems that would simplify the code a bit, what do you think? It does not block this PR to get merged

File: frontend/src/metabase/visualizations/visualizations/CartesianChart/events.ts
Comment (alxnddr) [inline frontend/src/metabase/visualizations/visualizations/CartesianChart/events.ts:705]: When there is only one element the footer seems excessive

**Review by alxnddr — APPROVED** [review]
Looks good! Please have a look at a couple of comments before merging

File: frontend/src/metabase/visualizations/visualizations/CartesianChart/events.ts
Comment (kulyk) [reply frontend/src/metabase/visualizations/visualizations/CartesianChart/events.ts]: Good catch! I ended up reusing the value from `chartModel.transformedDataset` [`5f7da16` (#48265)](https://github.com/metabase/metabase/pull/48265/commits/5f7da162f0cbeeae9274bf221d8d9e5406fbd406)

File: frontend/src/metabase/visualizations/visualizations/CartesianChart/events.ts
Comment (kulyk) [reply frontend/src/metabase/visualizations/visualizations/CartesianChart/events.ts]: Made it a last row instead [`fc13961` (#48265)](https://github.com/metabase/metabase/pull/48265/commits/fc139619006a8d378a468b5cfa529f8982d6ae25)

tooltips

**Pull request merged**
